component/libfuse/libfuse2_handler.go

```diff
@@ -440,8 +440,8 @@ func libfuse_create(path *C.char, mode C.mode_t, fi *C.fuse_file_info_t) C.int {
 		}
 	}
 
-	id := handlemap.Add(handle)
-	fi.fh = C.ulong(id)
+	handlemap.Add(handle)
+	fi.fh = C.ulong(uintptr(unsafe.Pointer(handle)))
```

Comment:
was there a reason we didn't go with this approach from the start? why did we go with the handlemap approach first for the fuse layer?